homework-selects/sql_questions.sql

```diff
@@ -3,55 +3,460 @@
+SELECT first_name, last_name, salary, department_id
+FROM employees
+WHERE department_id = (
+    SELECT department_id
+    FROM employees
+    WHERE employee_id = 201
+);
+
 --43. the first and last name, salary, and department ID for those employees whose salary is equal to the salary of the employee who works in that department which ID is 40.
+SELECT first_name, last_name, salary, department_id
+FROM employees
+WHERE salary IN (
+    SELECT salary
+    FROM employees
+    WHERE department_id = 40
+);
+
 --44. the first and last name, salary, and department ID for those employees who earn more than the minimum salary of a department which ID is 40.
+SELECT first_name, last_name, salary, department_id
+FROM employees
+WHERE salary > (
+    SELECT MIN(salary)
+    FROM employees
+    WHERE department_id = 40
+);
+
 --45. the first and last name, salary, and department ID for those employees who earn less than the minimum salary of a department which ID is 70.
+SELECT first_name, last_name, salary, department_id
+FROM employees
+WHERE salary > (
+    SELECT MIN(salary)
+    FROM employees
+    WHERE department_id = 40
+);
+
 --46. the first and last name, salary, and department ID for those employees who earn less than the average salary, and also work at the department where the employee Laura is working as a first name holder.
+SELECT first_name, last_name, salary, department_id
+FROM employees
+WHERE salary < (
+    SELECT AVG(salary)
+    FROM employees
+)
+  AND department_id = (
+    SELECT department_id
+    FROM employees
+    WHERE first_name = 'Laura'
+);
+
 --47. the full name (first and last name) of manager who is supervising 4 or more employees.
+SELECT (first_name || ' ' || last_name) AS employee_name
+FROM employees
+WHERE employee_id IN (
+    SELECT manager_id
+    FROM employees
+    GROUP BY manager_id
+    HAVING COUNT(manager_id) >= 4
+);
+
 --48. the details of the current job for those employees who worked as a Sales Representative in the past.
+SELECT *
+FROM jobs
+WHERE job_id IN (
+    SELECT job_id
+    FROM employees
+    WHERE employee_id IN(
+        SELECT DISTINCT employee_id
+        FROM job_history
+        WHERE job_id = (
+            SELECT job_id
+            FROM jobs
+            WHERE job_title = 'Sales Representative'
+        )
+    )
+);
+
 --49. all the infromation about those employees who earn second lowest salary of all the employees.
+SELECT *
+FROM employees
+WHERE salary IN (SELECT MIN(salary)
+                 FROM employees
+                 WHERE salary != (SELECT MIN(salary)
+                                  FROM employees)
+);
+
 --50. the department ID, full name (first and last name), salary for those employees who is highest salary drawar in a department.
+SELECT department_id, (first_name || ' ' || last_name) AS employee_name, salary
+FROM employees
+WHERE salary * commission_pct = (
+    SELECT MAX(salary * commission_pct)
+    FROM employees
+);
```

Comment:
SalaryDrawn is not related to commission_pct, hence, here you should do the following things:
1. Check if currentSalary of the employee is equal to result of a SUB-QUERY.
2. Sub-query should check if the department of the current employee is equal to the one we process in the main query (for this you can use aliases for tables), and also check if the department is not null.

Reply:
Done